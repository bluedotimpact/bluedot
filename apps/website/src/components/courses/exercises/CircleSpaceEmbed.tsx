import React, { CSSProperties } from 'react';

type CircleSpaceEmbedProps = {
  // Required
  spaceSlug: string;
  // Optional
  style?: CSSProperties;
};

const CircleSpaceEmbed: React.FC<CircleSpaceEmbedProps> = ({
  spaceSlug,
  style,
}) => {
  return (
    <iframe
      title={`Community discussion for ${spaceSlug}`}
      style={{
        border: '0',
        boxShadow: 'none',
        width: '800px',
        height: '80vh',
        ...style,
      }}
      src={`https://community.bluedot.org/c/${spaceSlug}?iframe=true`}
    />
  );
};

export default CircleSpaceEmbed;
