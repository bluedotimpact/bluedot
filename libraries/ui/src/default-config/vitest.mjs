/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
