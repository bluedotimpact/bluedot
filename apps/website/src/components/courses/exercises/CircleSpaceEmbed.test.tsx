import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import CircleSpaceEmbed from './CircleSpaceEmbed';

describe('CircleSpaceEmbed', () => {
  test('renders default as expected', () => {
    const { container } = render(
      <CircleSpaceEmbed
        spaceSlug="events"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with custom args', () => {
    const { container } = render(
      <CircleSpaceEmbed
        spaceSlug="events"
        style={{ width: '100%', height: '100%', minHeight: '510px' }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
