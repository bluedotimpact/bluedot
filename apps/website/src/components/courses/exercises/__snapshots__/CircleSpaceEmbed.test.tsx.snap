// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CircleSpaceEmbed > renders default as expected 1`] = `
<div>
  <iframe
    src="https://community.bluedot.org/c/events?iframe=true"
    style="border: 0px; box-shadow: none; width: 800px; height: 80vh;"
    title="Community discussion for events"
  />
</div>
`;

exports[`CircleSpaceEmbed > renders with custom args 1`] = `
<div>
  <iframe
    src="https://community.bluedot.org/c/events?iframe=true"
    style="border: 0px; box-shadow: none; width: 100%; height: 100%; min-height: 510px;"
    title="Community discussion for events"
  />
</div>
`;
